### PR TITLE
Keep up with quimb-1.5.0

### PR DESCRIPTION
**Pull request by pavoljuhas**
As of quimb-1.5.0 `TensorNetwork.contract(inplace=True)` returns
TensorNetwork instead of scalar.  Convert to scalar if needed.

### CODE REVIEW

**Review by vtomole — APPROVED** [review]
LGTM

**Comment by pavoljuhas** [comment]
@vtomole - actually the original CI failure was for a notebook tested with cirq head so it should have the fix in this PR.
This was caused by other notebook pip-installing released cirq; this interference should be resolved by 2f6b1a0d8973f1c9245307a01080a413c1356ccc.

The isolated_notebook_test.py tests w/r to the released cirq so I have disabled the affected notebook there.

File: cirq-core/cirq/contrib/quimb/state_vector.py [hunk 1]
```diff
@@ -177,6 +177,8 @@ def tensor_expectation_value(
     if ram_gb > max_ram_gb:
         raise MemoryError(f"We estimate that this contraction will take too much RAM! {ram_gb} GB")
     e_val = tn.contract(inplace=True)
+    if isinstance(e_val, qtn.TensorNetwork):
+        e_val = e_val.item()
```
Comment (viathor) [inline cirq-core/cirq/contrib/quimb/state_vector.py:181]: AFAICT, the contraction above will always produce a scalar, so why not call `item` unconditionally?

```python
e_val = tn.contract(inplace=True).item()
```

File: dev_tools/notebooks/notebook_test.py [hunk 2]
```diff
@@ -73,8 +73,11 @@ def require_packages_not_changed():
 def env_with_temporary_pip_target():
     """Setup system environment that tells pip to install packages to a temporary directory."""
     with tempfile.TemporaryDirectory(suffix='-notebook-site-packages') as tmpdirname:
+        # Note: We need to append the PYTHONPATH here so that development Cirq,
```
Comment (viathor) [inline dev_tools/notebooks/notebook_test.py:76]: nit: s/append/prepend/

File: cirq-core/cirq/contrib/quimb/state_vector.py (hunk 1 above)
Comment (pavoljuhas) [reply cirq-core/cirq/contrib/quimb/state_vector.py:181]: With quimb-1.4.2 `tn.contract()` may return a standard float which does not have the `item` method.
We could do `e_val = np.asarray(tn.contract(...)).item()`, but that is IMO harder to parse for humans than above. :)

We could also require `quimb==1.5`, but I feel that is a bit imposing on anyone installing cirq.

File: dev_tools/notebooks/notebook_test.py (hunk 2 above)
Comment (pavoljuhas) [reply dev_tools/notebooks/notebook_test.py:76]: Ack.  This test is run with `check/pytest` which sets PYTHONPATH to use the work-tree sources.
The line below appends `tmpdirname` to the PYTHONPATH.
When some notebook executes `pip install cirq`, that stable version would appear in PYTHONPATH after the worktree sources (which we want to use in this test).

Let me reword for clarity.

https://github.com/quantumlib/Cirq/blob/cc765a86064e7272047c427856ce89ba83953137/dev_tools/notebooks/notebook_test.py#L76-L83

**Review by viathor — APPROVED** [review]

**Pull request merged**